src/RoyaltyAutoClaim.sol

```diff
@@ -22,10 +24,26 @@ interface IRoyaltyAutoClaim {
 
     // Admin functions
     function updateReviewers(address[] memory _reviewers, bool[] memory _status) external;
-    function registerSubmission(string memory title, address royaltyRecipient) external;
-    function updateRoyaltyRecipient(string memory title, address newRoyaltyRecipient) external;
     function revokeSubmission(string memory title) external;
 
+    // Registration & Recipient Update functions
+    function registerSubmission(
+        string memory title,
+        address royaltyRecipient,
+        uint256[2] calldata _pA,
+        uint256[2][2] calldata _pB,
+        uint256[2] calldata _pC,
+        uint256[12] calldata _pubSignals
```

Comment:
Maybe encode them into a struct for better readability? e.g., ZKEmailProof
```
struct ZKEmailProof {
    uint256[2] a,
    ...
}
```

Reply:
Sure, will update it.